spec/unleash/constraint_spec.rb

```diff
@@ -384,23 +388,38 @@
       expect(constraint.matches_context?(context)).to be false
     end
 
-    it 'rejects constraint construction for invalid value types for operator' do
+    it 'gracefully handles invalid constraint operators' do
+      context_params = {
+        user_id: '123',
+        session_id: 'verylongsesssionid',
+        remote_address: '127.0.0.1',
+        properties: {
+          env: 'development'
+        }
+      }
+      context = Unleash::Context.new(context_params)
+      constraint = Unleash::Constraint.new('env', 'NOT_A_VALID_OPERATOR', 'dev', inverted: true)
+      expect(constraint.matches_context?(context)).to be true
+
+      constraint = Unleash::Constraint.new('env', 'NOT_A_VALID_OPERATOR', ['dev'], inverted: true)
+      expect(constraint.matches_context?(context)).to be true
```

Comment:
Maybe also test without the inverted clause?

```suggestion
      constraint = Unleash::Constraint.new('env', 'NOT_A_VALID_OPERATOR', 'dev')
      expect(constraint.matches_context?(context)).to be false

      constraint = Unleash::Constraint.new('env', 'NOT_A_VALID_OPERATOR', ['dev'])
      expect(constraint.matches_context?(context)).to be false

      constraint = Unleash::Constraint.new('env', 'NOT_A_VALID_OPERATOR', 'dev', inverted: true)
      expect(constraint.matches_context?(context)).to be true

      constraint = Unleash::Constraint.new('env', 'NOT_A_VALID_OPERATOR', ['dev'], inverted: true)
      expect(constraint.matches_context?(context)).to be true
```

or consider having just without the inverted parameter, so the test covers exactly it's objective and nothing else that is already tested elsewhere... 

```suggestion
      constraint = Unleash::Constraint.new('env', 'NOT_A_VALID_OPERATOR', 'dev')
      expect(constraint.matches_context?(context)).to be false

      constraint = Unleash::Constraint.new('env', 'NOT_A_VALID_OPERATOR', ['dev'])
      expect(constraint.matches_context?(context)).to be false
```

Feel free to pick between these two.

Reply:
done